pkgs/development/tools/tabnine/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchurl }:
+
+let
+  hashes = {
+    x86_64-linux   = "caa86a00d0ede8485bf855276ac1b2e411a5e99de612696b137ea3519a8e4967";
+    x86_64-darwin  = "eb5ceb534629dcf5d57640bf672ea0d94ac02e264209ab3a29e6bda8a44d3b8f";
```

Comment:
You can do something like this instead:

```nix
let
  platformSpecific = {
    x86_64-linux = {
      sha256 = "caa86a00d0ede8485bf855276ac1b2e411a5e99de612696b137ea3519a8e4967";
      string = "x86_64-apple-darwin";
    };
    # ...
  }.${stdenv.hostPlatform.system} or throw "Unsupported platform";
in
stdenv.mkDerivation {
  # ...
  src = fetchurl {
    # ...
    inherit (platformSpecific) sha256;
  };
```

And adjust the code below for this format.